### PR TITLE
Closes #2245 Added global key `core.https_allowed_credentials`

### LINKED ISSUE #2245: Option for Access-Control-Allow-Credentials header
# Required information

 * Distribution: Debian
 * Distribution version: Jessie
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.7.0
   * LXC version: 2.0.0
   * LXD version: 2.0.0
   * Storage backend in use: dir

# Issue description

[Allow web applications to run CORS](https://github.com/dobin/lxd-webgui/issues/5).
Firefox browser fails to make any request to lxd daemon api. It throws a `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at` error. 

# Steps to reproduce

 1. Install lxd on one machine
 2. Install lxd-webgui on the other
 3. Try to open it with Firefox Browser

# Information to attach

Can we have an option for `Access-Control-Allow-Credentials` header?

### PR DESCRIPTION
**Pull request by 18augst**
Original issue: https://github.com/dobin/lxd-webgui/issues/5

Updated my previous PR, added
`Signed-off-by: Nikolay Koshenkov <18augst@gmail.com>`
to commit message.

### CODE REVIEW

**Comment by stgraber** [comment]
As this adds a new configuration key, you also need to edit lxd/api_1.0.go and add an api extension entry, say "https_allowed_credentials" and then document it in doc/api_extensions.md.

**Comment by tych0** [comment]
Sorry, can you rebase this one again?

**Comment by 18augst** [comment]
Rebase is done.

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/api_1.0.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ func api10Get(d *Daemon, r *http.Request) Response {
 			"etag",
 			"patch",
 			"usb_devices",
+			"https_allowed_credentials"
```
Comment (stgraber) [inline lxd/api_1.0.go]: Missing trailing comma, this is causing the build to fail.

**Comment by 18augst** [comment]
I'm already beginning to feel uneasy. I hope that good will eventually come out more than the hassle. Really sorry for that.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**